parse/ast.go

```diff
@@ -0,0 +1,1146 @@
+package parse
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	"github.com/antlr4-go/antlr/v4"
+	"github.com/holiman/uint256"
+	"github.com/kwilteam/kwil-db/core/types"
+	"github.com/kwilteam/kwil-db/core/types/decimal"
+)
+
+// this file contains the ASTs for SQL, procedures, and actions.
+
+// Node is a node in the AST.
+type Node interface {
+	Accept(Visitor) any
+	GetNode() *Position
```

Comment:
I think `GetPosition` is better than `GetNode`

Reply:
Oh yeah. It was named Node and got changed to Position